ambry-store/src/main/java/com.github.ambry.store/DiskSpaceAllocator.java

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Utils;
+import java.io.File;
+import java.io.FileFilter;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * {@link DiskSpaceAllocator} handles the allocation of disk space to entities that require disk segments, such as the
+ * {@link BlobStore}s and {@link BlobStoreCompactor}s. The disk segments will be preallocated on platforms that support
+ * the fallocate syscall. The entities that require disk space register there requirements using the
+ * {@link #initializePool(Collection)} method. Disk space is allocated using the
+ * {@link DiskSpaceAllocator#allocate(File, long)} method. If a segment matching the requested size is not in the pool,
+ * it will be created at runtime. Disk space is returned to the pool using the
+ * {@link DiskSpaceAllocator#free(File, long)} method. It is up to the requester to keep track of the size of the
+ * segment requested.
+ *
+ * NOTE: Segments can be allocated and freed before pool initialization.  This will fall back to preallocating at
+ * runtime if the current pool does not include a segment of the correct size.
+ */
+class DiskSpaceAllocator {
+  private static final Logger LOGGER = LoggerFactory.getLogger(DiskSpaceAllocator.class);
```

Comment:
can we name this variable to "Logger" (which is the pattern we use across the board).